packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3462,12 +3483,19 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   @mustCallSuper
   void unmount() {
     assert(_debugLifecycleState == _ElementLifecycle.inactive);
-    assert(widget != null);
+    assert(_widget != null); // Use the private property to avoid a CastError during hot reload.
     assert(depth != null);
     assert(!_active);
-    final Key key = widget.key;
+    Key key;
```

Comment:
Can this just be:

```dart
final Key key = _widget.key;
```

Any reason we would want to go through the regular getter here?

Reply:
I suppose the only thing we're doing for is checking for global keys so it seems safe.